oioswift/common/middleware/copy.py

```diff
@@ -186,6 +187,54 @@ def _load_object_post_as_copy_conf(self, conf):
         except (NoSectionError, NoOptionError):
             pass
 
+    def allow_fast_copy(self, req):
+        """Check is fastcopy is possible."""
+
+        if req.method != 'PUT' or 'X-Copy-From' not in req.headers:
+            return False
+
+        # TODO maybe is it not necessary
+        already_checked = req.headers.get("Oio-Check-Fastcopy", 0)
+        if already_checked:
+            return False
+
+        self.logger.debug("COPY check fastcopy")
+
+        # mark request already checked
+        req.headers['Oio-Check-Fastcopy'] = 1
```

Comment:
It would be cleaner to add an entry to `req.environ`. I suggest `req.environ['oio.forbid_fast_copy'] = True` (other temporary metadata seem to use this format, e.g. `swift3.auth_details`).